src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -206,14 +115,165 @@ public boolean doMonitor(ServiceEmitter emitter)
+      }
+    }
+
+    void emit(ServiceEmitter emitter, ServiceMetricEvent.Builder builder)
+    {
+      builder.setDimension("gcGen", name);
+
+      if (collector != null) {
+        collector.emit(emitter, builder);
+      }
+
+      for (GcGenerationSpace space: spaces) {
+        space.emit(emitter, builder);
+      }
+    }
+  }
+
+  private class GcGenerationCollector
+  {
+    private String name;
+    private final LongCounter invocationsCounter;
+    private final TickCounter cpuCounter;
+    private long lastInvocations = 0;
+    private long lastCpuNanos = 0;
+
+    GcGenerationCollector(Map<String, JStatData.Counter<?>> jStatCounters, long genIndex)
+    {
+      String collectorKeyPrefix = String.format("sun.gc.collector.%d", genIndex);
+
+      String nameKey = String.format("%s.name", collectorKeyPrefix);
+      StringCounter nameCounter = (StringCounter) jStatCounters.get(nameKey);
+      name = getReadableName(nameCounter.getString());
+
+      invocationsCounter = (LongCounter) jStatCounters.get(String.format("%s.invocations", collectorKeyPrefix));
+      cpuCounter = (TickCounter) jStatCounters.get(String.format("%s.time", collectorKeyPrefix));
+    }
+
+    void emit(ServiceEmitter emitter, ServiceMetricEvent.Builder builder)
+    {
+      builder.setDimension("gcName", name);
+
+      long newInvocations = invocationsCounter.getLong();
+      emitter.emit(builder.build("jvm/gc/count", newInvocations - lastInvocations));
+      lastInvocations = newInvocations;
+
+      long newCpuNanos = cpuCounter.getNanos();
+      emitter.emit(builder.build("jvm/gc/cpu", newCpuNanos - lastCpuNanos));
+      lastCpuNanos = newCpuNanos;
+    }
+
+    private String getReadableName(String name)
+    {
+      switch (name) {
+        // Young gen
+        case "Copy": return "serial";
+        case "PSScavenge": return "parallel";
+        case "PCopy": return "cms";
+        case "G1 incremental collections": return "g1";
+
+        // Old gen
+        case "MCS": return "serial";
+        case "PSParallelCompact": return "parallel";
+        case "CMS": return "cms";
+        case "G1 stop-the-world full collections": return "g1";
+
+        default: return name;
+      }
+    }
+  }
+
+  private class GcGenerationSpace
+  {
+    private String name;
+
+    private LongCounter maxCounter;
+    private LongCounter capacityCounter;
+    private LongCounter usedCounter;
+    private LongCounter initCounter;
+
+    GcGenerationSpace(Map<String, JStatData.Counter<?>> jStatCounters, long genIndex, long spaceIndex)
+    {
+      String spaceKeyPrefix = String.format("sun.gc.generation.%d.space.%d", genIndex, spaceIndex);
+
+      String nameKey = String.format("%s.name", spaceKeyPrefix);
+      StringCounter nameCounter = (StringCounter) jStatCounters.get(nameKey);
+      name = nameCounter.toString().toLowerCase();
+
+      maxCounter      = (LongCounter) jStatCounters.get(String.format("%s.maxCapacity",  spaceKeyPrefix));
```

Comment:
Do we have a practice of aligning code like this? Personally I'm not a fan, because if another line is added, the alignment in the whole block might have to be changed, that makes git diff bigger, and also people usually forget to update.